### ISSUE TITLE
problem in decommission

### ISSUE THREAD

**Issue by harelassayag** [scylladb/scylla-cluster-tests#349]
we get an exception in[ get_node_info_list](https://github.com/scylladb/scylla-cluster-tests/blob/master/sdcm/nemesis.py#L222). then, we don't destroy the node and don't add another node instead.

```

017-08-09 14:42:27,634 remote           L0197 DEBUG| [10.240.0.95] [stdout] Aug 09 14:42:27 info   |  [shard 0] storage_service - DECOMMISSIONING: done
2017-08-09 14:42:27,641 remote           L0197 DEBUG| [10.240.0.95] [stderr] Connection to 10.240.0.95 closed.
2017-08-09 14:42:27,642 nemesis          L0107 DEBUG| sdcm.nemesis.ChaosMonkey: Command '/bin/ssh -tt -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpI1YdUa -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l scylla-test -p 22 -i /jenkins/.ssh/scylla-test 10.240.0.95 "nodetool --host localhost decommission"' duration -> 303.728873014 s
2017-08-09 14:42:27,642 remote           L0894 DEBUG| Remote [scylla-test@10.240.0.87]: Running 'nodetool status'
2017-08-09 14:42:27,642 remote           L0097 INFO | [10.240.0.87] Running '/bin/ssh -tt -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpFV30Jo -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l scylla-test -p 22 -i /jenkins/.ssh/scylla-test 10.240.0.87 "nodetool status"'
2017-08-09 14:42:27,955 remote           L0197 DEBUG| [10.240.0.87] [stdout] bash: nodetool: command not found
2017-08-09 14:42:27,955 remote           L0197 DEBUG| [10.240.0.87] [stderr] Connection to 10.240.0.87 closed.
2017-08-09 14:42:27,956 nemesis          L0399 ERROR| sdcm.nemesis.ChaosMonkey: Unhandled exception in method <function disrupt at 0x58e56e0>
Traceback (most recent call last):
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/nemesis.py", line 396, in wrapper
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/nemesis.py", line 519, in disrupt
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/nemesis.py", line 352, in call_random_disrupt_method
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/nemesis.py", line 223, in disrupt_nodetool_decommission
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/cluster.py", line 1686, in get_node_info_list
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/remote.py", line 905, in run
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/remote.py", line 849, in _run
  File "/jenkins/slave/workspace/scylla-2.0-longevity-gce/label/gce/sdcm/remote.py", line 228, in ssh_run
CmdError: Command '/bin/ssh -tt -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpFV30Jo -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l scylla-test -p 22 -i /jenkins/.ssh/scylla-test 10.240.0.87 "nodetool status"' failed (rc=127)
2017-08-09 14:42:27,956 nemesis          L0407 DEBUG| sdcm.nemesis.ChaosMonkey: Decommission Node longevity-scylla-2-0-scylla-db-node-7b8f147e-0-2 [104.196.116.195 | 10.240.0.95] (seed: False) duration -> 304 s
2017-08-09 14:42:27,957 remote           L0894 DEBUG| Remote [scylla-test@10.240.0.28]: Running 'nodetool status'
2017-08-09 14:42:27,957 remote           L0097 INFO | [10.240.0.28] Running '/bin/ssh -tt -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpk43hwp -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l scylla-test -p 22 -i /jenkins/.ssh/scylla-test 10.240.0.28 "nodetool status"'
2017-08-09 14:42:28,974 remote           L0197 DEBUG| [10.240.0.52] [stdout] Aug 09 14:42:28 info   |  [shard 4] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20404-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20324-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20292-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20436-Data.db:level=0, ]
2017-08-09 14:42:30,722 remote           L0197 DEBUG| [10.240.0.99] [stdout] total,    7018647158,   71534,   71534,   71534,    14.0,    12.0,    30.6,    47.0,    64.1,   110.1,101661.5,  0.00062,      0,      0,       0,       0,       0,       0
2017-08-09 14:42:30,939 remote           L0197 DEBUG| [10.240.0.52] [stdout] Aug 09 14:42:30 info   |  [shard 13] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20301-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20365-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20157-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-405a83307c2411e79abe000000000000/keyspace1-standard1-ka-20253-Data.db:level=0, ]
2017-08-09 14:42:31,177 remote           L0197 DEBUG| [10.240.0.28] [stdout] Datacenter: datacenter1
2017-08-09 14:42:31,177 remote           L0197 DEBUG| [10.240.0.28] [stdout] =======================
2017-08-09 14:42:31,177 remote           L0197 DEBUG| [10.240.0.28] [stdout] Status=Up/Down
2017-08-09 14:42:31,179 remote           L0197 DEBUG| [10.240.0.28] [stdout] |/ State=Normal/Leaving/Joining/Moving
2017-08-09 14:42:31,179 remote           L0197 DEBUG| [10.240.0.28] [stdout] --  Address      Load       Tokens       Owns    Host ID                               Rack
2017-08-09 14:42:31,190 remote           L0197 DEBUG| [10.240.0.28] [stdout] UN  10.240.0.96  86.56 GB   256          ?       9f571f12-92d0-4523-b34b-e2b2e65a46e4  rack1
2017-08-09 14:42:31,196 remote           L0197 DEBUG| [10.240.0.28] [stdout] UN  10.240.0.52  94.11 GB   256          ?       c6b306aa-68fc-416a-997c-725d4b6c5e6f  rack1
2017-08-09 14:42:31,204 remote           L0197 DEBUG| [10.240.0.28] [stdout] UN  10.240.0.28  72.63 GB   256          ?       ec8a4519-ebc0-48b0-93da-dad314723579  rack1
2017-08-09 14:42:31,204 remote           L0197 DEBUG| [10.240.0.28] [stdout] 
:

```

**Comment by roydahan** [comment]
@abvgedeika, please take a look on this and try to fix it ASAP.
We should try catching these kind of errors, produce a warning and try to run it on a different node.

**Comment by amoskong** [comment]
The error raised from the verification_node(10.240.0.87) is `[stdout] bash: nodetool: command not found`. Is it mean the node setup didn't complete rightly?

**Comment by roydahan** [comment]
I don't think it something related to setup.
When node is down, the nodetool status doens't work on it.
Why the node was down, is a different question, but we should be aware of such cases and handle them better (e.g. retry with another node).

**Comment by amoskong** [comment]
Solved by commit 14a11b12989f28527179ce5965ab897e4cda9e5a
